default/AI/ResearchAI.py

```diff
@@ -1,30 +1,328 @@
+
+    # Organic hulls
+    if tech_name in ["SHP_ORG_HULL", "SHP_SENT_HULL"]:
+        return org
+    if tech_name in ["SHP_MULTICELL_CAST", "SHP_ENDOCRINE_SYSTEMS", "SHP_CONT_BIOADAPT"]:
+        return min(org, orgraven)
+    if tech_name in ["SHP_MONOCELL_EXP", "SHP_CONT_SYMB", "SHP_BIOADAPTIVE_SPEC"]:
+        return min(org, orgneural)
+    if tech_name in ["SHP_ENDOSYMB_HULL"]:
+        return min(org, orgendosym)
+
+    # energy hulls
+    if tech_name in ["SHP_FRC_ENRG_COMP"]:
+        return energy
+    if tech_name in ["SHP_QUANT_ENRG_MAG"]:
+        return min(energy, energymag, immediate if bright_star or blackhole else useless)
+    if tech_name in ["SHP_ENRG_BOUND_MAN"]:
+        return min(energy, energyfrac, immediate if bright_star or blackhole else useless)
+    if tech_name in ["SHP_SOLAR_CONT"]:
+        return min(energy, immediate if red_star or blackhole else useless) # red star can be turned into blackhole by Artificial black hole
+
+    # damage control
+    if tech_name in ["SHP_REINFORCED_HULL"]:
+        return default
+    if tech_name in ["SHP_BASIC_DAM_CONT", "SHP_FLEET_REPAIR", "SHP_ADV_DAM_CONT"]:
+        return repair
+
+    # ship parts
+    if tech_name in ["SHP_IMPROVED_ENGINE_COUPLINGS", "SHP_N_DIMENSIONAL_ENGINE_MATRIX", "SHP_SINGULARITY_ENGINE_CORE"]:
+        return engine
+    if tech_name in ["SHP_DEUTERIUM_TANK", "SHP_ANTIMATTER_TANK", "SHP_ZERO_POINT"]:
+        return fuel
+    if tech_name in ["SHP_NOVA_BOMB", "SHP_DEATH_SPORE", "SHP_BIOTERM"]:
+        return useless # AI does not use Nova bomb nor bioterror
+
+    # ship shields
+    if tech_name in ["SHP_DEFLECTOR_SHIELD", "SHP_PLASMA_SHIELD", "SHP_BLACKSHIELD"]:
+        return shield
+    if tech_name in ["SHP_MULTISPEC_SHIELD"]:
+        return stealth
+
+    # ship armor
+    if tech_name in ["SHP_ROOT_ARMOR", "SHP_ZORTRIUM_PLATE"]:
+        return armor
+    if tech_name in ["SHP_DIAMOND_PLATE", "SHP_XENTRONIUM_PLATE"]:
+        return useless if asteroid or neutron else armor # asteroid hull line and neutronium extraction includes better armors
+
+    # weapons
+    if tech_name in ["SHP_ROOT_AGGRESSION", "SHP_WEAPON_1_2", "SHP_WEAPON_1_3", "SHP_WEAPON_1_4"]:
+        return massdriver and not tech_is_complete("SHP_WEAPON_4_1") # don't research obsolete weapons if get deathray from ruins
+    if tech_name in ["SHP_WEAPON_2_1", "SHP_WEAPON_2_2", "SHP_WEAPON_2_3", "SHP_WEAPON_2_4"]:
+        return laser and not tech_is_complete("SHP_WEAPON_4_1")
+    if tech_name in ["SHP_WEAPON_3_1", "SHP_WEAPON_3_2", "SHP_WEAPON_3_3", "SHP_WEAPON_3_4"]:
+        return plasmacannon and not tech_is_complete("SHP_WEAPON_4_1")
+    if tech_name in ["SHP_WEAPON_4_1", "SHP_WEAPON_4_2", "SHP_WEAPON_4_3", "SHP_WEAPON_4_4"]:
+        return deathray
+
+    # detection
+    if tech_name in ["SPY_DETECT_1", "SPY_DETECT_2", "SPY_DETECT_3", "SPY_DETECT_4", "SPY_DETECT_5", "SPY_DIST_MOD", "SPY_LIGHTHOUSE"]:
+        return detection
```

Comment:
Maybe prioritize the first (not preunlocked) detection tech a little bit more to find stealthed monsters, Laenfa etc.